src/charm.py

```diff
@@ -866,7 +912,12 @@ def _on_leader_elected(self, event: LeaderElectedEvent) -> None:
             PATRONI_PASSWORD_KEY,
         }:
             if self.get_secret(APP_SCOPE, password) is None:
-                self.set_secret(APP_SCOPE, password, new_password())
+                if password in system_user_passwords:
+                    # use provided passwords for system-users if available
+                    self.set_secret(APP_SCOPE, password, system_user_passwords[password])
+                else:
+                    # generate a password for this user if not provided
+                    self.set_secret(APP_SCOPE, password, new_password())
```

Comment:
We should most probably log a message here about which password was autogenerated and which was taken from the secret. If the user makes a typo in the key, there will be no way to know that the charm generated its own password.